src/classes/Database.php

```diff
@@ -447,4 +447,104 @@ public function getUsername(int $userId): string|false {
             throw $e;
         }
     }
+
+    /**
+     * @return array{username: string, name: ?string, email: ?string}|false
+     */
+    public function getUserProfile(int $userId): array|false {
+        Logger::debug("Getting user profile", ['user_id' => $userId]);
+
+        if ($this->pdo === null) {
+            $this->connect();
+        }
+
+        if ($this->pdo === null) {
+            throw new \RuntimeException("Failed to connect to database");
+        }
+
+        try {
+            $stmt = $this->pdo->prepare("SELECT username, name, email FROM users WHERE id = ?");
+            $stmt->execute([$userId]);
+            $row = $stmt->fetch(\PDO::FETCH_ASSOC);
+
+            if ($row === false || !is_array($row)) {
+                Logger::warning("User not found when fetching profile", ['user_id' => $userId]);
+                return false;
+            }
+
+            return [
+                'username' => (string)$row['username'],
+                'name' => $row['name'] !== null ? (string)$row['name'] : null,
+                'email' => $row['email'] !== null ? (string)$row['email'] : null,
```

Comment:
@jetbrains-junie use a model (e.g. `UserProfileData`) here,  instead of an array. Update related code as well, to work based on the model, instead of an array.